### PR TITLE
fix: database base ttl

### LINKED ISSUE #4914: Database options do not take effect
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Other

### Minimal reproduce step

1. Create a database with ttl
```bash
mysql -h 127.0.0.1 -P 4002 -e "create database benchmark with (ttl='1d');"
```

2. Create a table in that database
```bash
mysql -h 127.0.0.1 -P 4002 benchmark -e "CREATE TABLE IF NOT EXISTS cpu ( hostname STRING NULL, region STRING NULL, datacenter STRING NULL, rack STRING NULL, os STRING NULL, arch STRING NULL, team STRING NULL, service STRING NULL, service_version STRING NULL, service_environment STRING NULL, usage_user BIGINT NULL, usage_system BIGINT NULL, usage_idle BIGINT NULL, usage_nice BIGINT NULL, usage_iowait BIGINT NULL, usage_irq BIGINT NULL, usage_softirq BIGINT NULL, usage_steal BIGINT NULL, usage_guest BIGINT NULL, usage_guest_nice BIGINT NULL, ts TIMESTAMP(9) NOT NULL, TIME INDEX (ts), PRIMARY KEY (hostname, region, datacenter, rack, os, arch, team, service, service_version, service_environment));"
```

### What did you expect to see?

The table (and regions in that table) created should inherit the database `ttl` option

### What did you see instead?

The ttl option does not take effect in region level.

### What operating system did you use?

NA

### What version of GreptimeDB did you use?

0.9.5

### Relevant log output and stack trace

In #2341 we introduced procedure framework for fault-tolerant DDL tasks like creating tables. create table procedure are split into several steps, namely creating table metadata and invoking datanodes to create regions. 

At the [first step](https://github.com/GreptimeTeam/greptimedb/blob/fd8eba36a8d03d7ce7b9566ebc17220d728f9fd5/src/operator/src/statement/ddl.rs#L1313), table metadata inherits the database level options. However, the merge result is not reflected in `CreateTableExpr` itself. So at the second step, which is invoking datanodes to create regions, the `CreateTableExpr` does not carry any database-level options:

https://github.com/GreptimeTeam/greptimedb/blob/6a634f8e5d870e7dd9000eb68ce30a936d09fb9c/src/common/meta/src/ddl/create_table.rs#L170

So the created regions do not contain schema-level options.

This bug is elusive in that you cannot find that schema options is not applied to regions since all table metadata are correct, so `SHOW CREATE TABLE` statement can give the correct result.

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
- closes #4914

## What's changed and what's your intention?

Fix the issue that database options are not reflected in table. 

Main changes:
- stop merging database options into table options when creating table https://github.com/GreptimeTeam/greptimedb/pull/4926/files#diff-2563d4135f7e232aaec6a39204e6e9afc21801dff0c957772ce4dd7d3bd271d4L1313
- when calculating ttl, query from metasrv for database ttl in place 
- when executing `SHOW CREATE TABLE` statements, query from metasrv for table actual TTL options.
  - this will also affect `CREATE TABLE ... LIKE` statements. If table A inherits database level TTL, then `CREATE TABLE b like a` will also inherit that TTL statically.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/query/src/sql/show_create_table.rs [hunk 1]
```diff
@@ -47,7 +49,12 @@ fn create_sql_options(table_meta: &TableMeta) -> OptionMap {
     }
     if let Some(ttl) = table_opts.ttl {
         options.insert(TTL_KEY.to_string(), format_duration(ttl).to_string());
-    }
+    } else if let Some(database_ttl) = schema_options.and_then(|o| o.ttl) {
```
Comment (MichaelScofield) [inline src/query/src/sql/show_create_table.rs:52]: Kind of "gild the lily" to me here. I think if the table is created without the ttl options, then the "show create" is supposed to show its creation as is, no need to add any schema-level options to it.

File: src/mito2/src/compaction.rs [hunk 2]
```diff
@@ -256,10 +262,23 @@ impl CompactionScheduler {
...
         } = request;
+
+        let ttl = find_ttl(
+            region_id.table_id(),
+            current_version.options.ttl,
+            &schema_metadata_manager,
+        )
+        .await
+        .unwrap_or_else(|e| {
+            warn!(e; "Failed to get ttl for region: {}", region_id);
```
[the remark this answers is not in the document]
Comment (v0y4g3r) [reply src/mito2/src/compaction.rs:275]: > Why not return the error, is failing to fetch the correct database ttl not important for doing compaction?

No, temporarily metasrv failure should not fail a compaction task

File: src/query/src/sql/show_create_table.rs (hunk 1 above)
Comment (v0y4g3r) [reply src/query/src/sql/show_create_table.rs:52]: > Kind of "gild the lily" to me here. I think if the table is created without the ttl options, then the "show create" is supposed to show its creation as is, no need to add any schema-level options to it.

I discussed this with @fengjiachun and @WenyXu . If the database TTL actually takes effect, it should be reflected in SHOW CREATE TABLE statement.

File: src/mito2/src/compaction/compactor.rs [hunk 3]
```diff
@@ -169,6 +172,12 @@ pub async fn open_compaction_region(
         Arc::new(version)
     };
 
+    let ttl = find_ttl(
+        req.region_id.table_id(),
+        current_version.options.ttl,
+        &schema_metadata_manager,
+    )
+    .await?;
```
[the remark this answers is not in the document]
Comment (v0y4g3r) [reply src/mito2/src/compaction/compactor.rs]: > Is this an inconsistent behavior to the same `find_ttl` call in https://github.com/GreptimeTeam/greptimedb/pull/4926/files#diff-f413376968d6fea8034a8259e2f933c1cdfc370f30fa0d49df8a63156bb9cd9bR274?

You're right, we should also ignore the failure here.

File: src/mito2/src/compaction.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichaelScofield) [reply src/mito2/src/compaction.rs:275]: Then will the compaction produce incorrect results if the ttl is none while it shouldn't be?

File: src/mito2/src/compaction/compactor.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (v0y4g3r) [reply src/mito2/src/compaction/compactor.rs]: > Is this an inconsistent behavior to the same `find_ttl` call in https://github.com/GreptimeTeam/greptimedb/pull/4926/files#diff-f413376968d6fea8034a8259e2f933c1cdfc370f30fa0d49df8a63156bb9cd9bR274?

fixed in d163055c2c

**Review by MichaelScofield — APPROVED** [review]

File: src/mito2/src/compaction.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (v0y4g3r) [reply src/mito2/src/compaction.rs:275]: > Then will the compaction produce incorrect results if the ttl is none while it shouldn't be?

No, if ttl is none, expired ssts will not be removed.

File: src/mito2/src/engine.rs [hunk 4]
```diff
@@ -112,13 +113,15 @@ impl MitoEngine {
         mut config: MitoConfig,
         log_store: Arc<S>,
         object_store_manager: ObjectStoreManagerRef,
+        kv_backend: KvBackendRef,
```
Comment (WenyXu) [inline src/mito2/src/engine.rs:116]: better to use `SchemaMetadataManager`? cc @evenyag 
```suggestion
        schema_metadata_manager: SchemaMetadataManager
```

**Review by WenyXu — APPROVED** [review]
Rest LGTM

File: src/mito2/src/engine.rs (hunk 4 above)
Comment (v0y4g3r) [reply src/mito2/src/engine.rs]: fixed in https://github.com/GreptimeTeam/greptimedb/pull/4926/commits/6a82a59367c8960c81274a973f8d963b73b47b90

File: src/mito2/src/compaction.rs [hunk 5]
```diff
@@ -414,6 +434,24 @@ impl PendingCompaction {
...
+    table_id: TableId,
+    table_ttl: Option<Duration>,
+    schema_metadata_manager: &SchemaMetadataManagerRef,
+) -> Result<Option<Duration>> {
+    if let Some(table_ttl) = table_ttl {
+        return Ok(Some(table_ttl));
+    }
+
+    let ttl = schema_metadata_manager
+        .get_schema_options_by_table_id(table_id)
```
Comment (fengjiachun) [inline src/mito2/src/compaction.rs:448]: Consider a timeout setting to prevent the engine from being blocked by slow network I/O.

**Pull request merged**